### PR TITLE
Fix RSOCtoPSD bridge for dimension 2

### PR DESCRIPTION
**Pull request by blegat**
With this `MultiSetMapBridge`, it should be easy to fix other similar case, e.g., `SecondOrderCone(1)` -> `Nonnegatives(1)` while `SecondOrderCone(2)` -> `RotatedSecondOrderCone(2)` but we can wait for someone or for a test (like in this case) requesting to fix these corner cases before doing it.

### CODE REVIEW

**Comment by odow** [comment]
I've hit this a few times, and always seen this choice as intentional.

File: test/Bridges/Constraint/soc_to_psd.jl [hunk 1]
```diff
@@ -146,28 +146,31 @@ function test_RSOCtoPSD()
...
+                MOI.PositiveSemidefiniteConeTriangle,
+                0,
+            ),
+            (MOI.VectorAffineFunction{Float64}, MOI.Nonnegatives, 0),
+        ),
     )
     return
 end
 
-# TODO(odow): we should fix this so it does not error.
```
Comment (odow) [inline test/Bridges/Constraint/soc_to_psd.jl:158]: I guess perhaps not intentional. I like seeing TODOs from the past.

**Comment by blegat** [comment]
Yes I'm also hitting this often. When you have a code generating socs of various dimensions and one if them is t^2 >= x^2, it is bridged to RSOC and it tries to bridge from RSOC to PSD and you get this error. Anothet use case of this MultiSetMap is to have faster reformulations. For instance, in Sumofsquares, I reformulate 2x2 psd into socs instead of psd since it's faster

File: src/Bridges/Constraint/bridges/soc_to_psd.jl [hunk 2]
```diff
@@ -208,13 +237,27 @@ function MOI.Bridges.inverse_map_set(
...
 end
 
+function MOI.Bridges.inverse_map_set(
+    ::Type{<:RSOCtoPSDBridge},
+    ::MOI.Nonnegatives,
+)
+    return MOI.RotatedSecondOrderCone(2)
+end
+
+function _rsoc_to_psd_too_small_message(n)
```
Comment (odow) [inline src/Bridges/Constraint/bridges/soc_to_psd.jl:247]: Remove

File: src/Bridges/Constraint/set_map.jl [hunk 3]
```diff
@@ -5,91 +5,98 @@
...
-end
-
 # Attributes, Bridge acting as a model
 
-function MOI.get(
-    ::SetMapBridge{T,S2,S1,F},
-    ::MOI.NumberOfConstraints{F,S2},
-)::Int64 where {T,S2,S1,F}
-    return 1
+_get(::MOI.ConstraintIndex, ::MOI.NumberOfConstraints) = 0
```
Comment (odow) [inline src/Bridges/Constraint/set_map.jl:71]: I've found that these "helper" functions just make things more difficult in the long run. Let's just explicitly write out the implementation for both constraint types.

**Comment by odow** [comment]
Are you up late working on this? Otherwise I will push forward this afternoon

**Comment by odow** [comment]
https://github.com/jump-dev/MathOptInterface.jl/actions/runs/7204367333

**Review by odow — APPROVED** [review]
